x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -16,104 +16,242 @@ import {
+  const dispatch = useDispatch();
+  const onSearchChange = useCallback(
+    (value: string) => {
+      dispatch({ type: 'search:update', value });
+    },
+    [dispatch]
+  );
+
+  const searchTerm = search.term.trim();
+
+  const jsonBlock = (
+    <EuiCodeBlock
+      language="json"
+      isCopyable
+      data-test-subj="indexDetailsMappingsCodeBlock"
       css={css`
         height: 100%;
       `}
     >
-      <EuiFlexItem
-        grow={1}
-        css={css`
-          min-width: 400px;
-        `}
-      >
-        <EuiPanel grow={false} paddingSize="l">
-          <EuiFlexGroup alignItems="center" gutterSize="s">
-            <EuiFlexItem grow={false}>
-              <EuiIcon type="iInCircle" />
-            </EuiFlexItem>
-            <EuiFlexItem>
-              <EuiTitle size="xs">
-                <h2>
-                  <FormattedMessage
-                    id="xpack.idxMgmt.indexDetails.mappings.docsCardTitle"
-                    defaultMessage="About index mappings"
-                  />
-                </h2>
-              </EuiTitle>
-            </EuiFlexItem>
-          </EuiFlexGroup>
-          <EuiSpacer size="s" />
-          <EuiText>
-            <p>
-              <FormattedMessage
-                id="xpack.idxMgmt.indexDetails.mappings.docsCardDescription"
-                defaultMessage="Your documents are made up of a set of fields. Index mappings give each field a type
-              (such as keyword, number, or date) and additional subfields. These index mappings determine the functions
-              available in your relevance tuning and search experience."
-              />
-            </p>
-          </EuiText>
-          <EuiSpacer size="m" />
-          <EuiLink
-            data-test-subj="indexDetailsMappingsDocsLink"
-            href={documentationService.getMappingDocumentationLink()}
-            target="_blank"
-            external
-          >
-            <FormattedMessage
-              id="xpack.idxMgmt.indexDetails.mappings.docsCardLink"
-              defaultMessage="Learn more about mappings"
-            />
-          </EuiLink>
-        </EuiPanel>
-        {extensionsService.indexMappingsContent && (
-          <>
-            <EuiSpacer />
-            {extensionsService.indexMappingsContent.renderContent({ index, getUrlForApp })}
-          </>
+      {data}
+    </EuiCodeBlock>
+  );
+
+  const treeViewBlock = (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        {mappingsDefinition === null ? (
+          <EuiEmptyPrompt
+            color="danger"
+            iconType="error"
+            title={
+              <h2>
+                <FormattedMessage
+                  id="xpack.idxMgmt.indexDetails.mappings.invalidMappingKeysErrorMessageTitle"
+                  defaultMessage="Unable to load the mapping"
+                />
+              </h2>
+            }
+            body={
+              <h2>
```

Comment:
Can we use a smaller font for the message body (I think `<p>` would be better than `<h2>`)?
Also, I'm not sure if we should add the part "Please provide a mapping with valid keys." given that the index with these mappings could be auto-generated as is the case with the indices that caused the blank pages.

@MichaelMarcialis could you let us know if you have any suggestions on the design/copy of the error prompt?

<img width="1704" alt="Screenshot 2024-03-18 at 16 18 46" src="https://github.com/elastic/kibana/assets/59341489/1d566c31-5e80-442b-b329-24b73786978b">

Reply:
Thanks, @ElenaStoeva. I agree that using a paragraph `p` element in the `EuiEmptyPrompt`'s body would look better visually while also being more semantically correct.

As for the "Please provide a mapping with valid keys." sentence, if I were the user, I'd expect this error to provide some guidance as to how or where I could go to fix the situation. Is there a page or action we could guide the user towards via a button/link in the empty prompt? How would a user typically remedy this situation (in both auto and manually generated scenarios)?

Finally, regarding general design suggestions, I'd also consider making the empty prompt the only element on the page below the tab interface. The empty prompt looks odd when nested inside another panel, and it isn't helpful for us to offer the user a field search and index mapping description when it is in an erroneous state like this.